### DISCUSSION TITLE
Custom cross reference schemes

### DISCUSSION THREAD

**Discussion by sorenoneill** [quarto-dev/quarto-cli#4343] [Feature Requests]
I needed to count figures in main text and supplementary materials separately -- solved this with a dirty hack by hijacking the theorem scheme (see below).

A better solution would be the ability to create custom cross reference schemes as children (inheritance) from existing once.

For example a new Figure based scheme:

```{r}
---
title: "A title"
crossref:
  new_scheme:
    inherits: Figure
    name: supplementary-figure
    title:  Supplementary figure
    prefix: Supplementary figure
    labels: alpha A
---

## My markdown document

...with some meaningful content.
```

<hr>

My dirty hack 

``` {r}
---
title: "Figure cross-reference resetting"
subtitle: "A dirty hack"
format: html
crossref:
  thm-title:  Supplementary figure
  thm-prefix: Supplementary figure
  thm-labels: alpha A
---

## A manuscript

See @fig-elephant

See @thm-supplementary-elephant in the supplementary materials.

## Figures

![An elephant](elephant.png){#fig-elephant}

## Supplementary materials

::: {#thm-supplementary-elephant}
![The same elephant but now as supplementary material](elephant.png)
:::
```

**Comment by mcanouil** [comment]
Closing in favour of https://github.com/quarto-dev/quarto-cli/issues/4944

**Reply by baarthur** [reply]
I'd just add an additional request: in addition to child categories, it would be nice to have separate lists too. E.g. my university standards requires us to label figures, maps, and graphs separately when there are more than 5, and in that case we need to include also a list of maps, list of graphs and so on.

**Reply by mcanouil** [reply]
@baarthur This was closed in favour of the mentioned issue.
There is no point to post here.
Also as given in the other discussion, see <https://quarto.org/docs/prerelease/1.4/crossref.html#custom-crossreferenceable-types>.

**Reply by baarthur** [reply]
i'm sorry i'm still kind of lost here ---I thought that https://github.com/quarto-dev/quarto-cli/issues/4944 was a to-do list and not a specific issue.

**Reply by cscheid** [reply]
It tracks all of the changes I'm working on, and that includes lists (they're called collations in #4944 but they're there).

**Reply by mcanouil** [reply]
> i'm sorry i'm still kind of lost here ---I thought that https://github.com/quarto-dev/quarto-cli/issues/4944 was a to-do list and not a specific issue.

The "issue" with posting on a closed/outdated discussion is that it might not get attention.
We try to indicate where to track the issue or continue the discussion "closed in favour of" for instance.

When you see something saying to go elsewhere, follow the link:
1. To check if what you are going to ask is not already there.
2. To add something that might be useful.

Or open a new discussion mentioning other discussions/issues (closed or opened if related but not a duplicate).
